skrifa/src/scale/mod.rs

```diff
@@ -1,13 +1,128 @@
+//! # }
+//! ```
+//!
+//! For variable fonts, the [`variation_settings`](ScalerBuilder::variation_settings) method can
+//! be used to specify user coordinates for selecting an instance:
+//!
+//! ```
+//! # use skrifa::{scale::*, Size};
+//! # fn build_scaler(font: read_fonts::FontRef) {
+//! let mut context = Context::new();
+//! let mut scaler = context.new_scaler()
+//!     .size(Size::new(16.0))
+//!     .variation_settings(&[("wght", 720.0), ("wdth", 75.0)])
+//!     .build(&font);
+//! # }
+//! ```
+//!
+//! If you already have coordinates in normalized design space, you can specify those directly
+//! with the [`normalized_coords`](ScalerBuilder::normalized_coords) method.
+//!
+//! See the [`ScalerBuilder`] type for all available configuration options.
+//!
+//! ## Getting an outline
+//!
+//! Once we have a configured scaler, extracting an outline is fairly simple. The
+//! [`outline`](Scaler::outline) method on scaler uses a callback approach where the user
+//! provides an implementation of the [`Pen`] trait and the appropriate methods are invoked for
+//! each resulting path element of the scaled outline.
+//!
+//! Assuming we constructed a scaler as above, let's load a glyph and convert it into an SVG path:
+//!
+//! ```
+//! # use skrifa::{scale::*, GlyphId, Size};
+//! # fn build_scaler(font: read_fonts::FontRef) {
+//! # let mut context = Context::new();
+//! # let mut scaler = context.new_scaler()
+//! #    .size(Size::new(16.0))
+//! #    .build(&font);
+//! // Create a type for holding our SVG path.
+//! #[derive(Default)]
+//! struct SvgPath(String);
+//!
+//! // Implement the Pen trait for this type. This emits the appropriate
+//! // SVG path commands for each element type.
+//! impl Pen for SvgPath {
+//!     fn move_to(&mut self, x: f32, y: f32) {
+//!         self.0.push_str(&format!("M{x:.1},{y:.1} "));
+//!     }
+//!
+//!     fn line_to(&mut self, x: f32, y: f32) {
+//!         self.0.push_str(&format!("L{x:.1},{y:.1} "));
+//!     }
+//!
+//!     fn quad_to(&mut self, cx0: f32, cy0: f32, x: f32, y: f32) {
+//!         self.0
+//!             .push_str(&format!("Q{cx0:.1},{cy0:.1} {x:.1},{y:.1} "));
+//!     }
+//!
+//!     fn curve_to(&mut self, cx0: f32, cy0: f32, cx1: f32, cy1: f32, x: f32, y: f32) {
+//!         self.0.push_str(&format!(
+//!             "C{cx0:.1},{cy0:.1} {cx1:.1},{cy1:.1} {x:.1},{y:.1} "
+//!         ));
+//!     }
+//!
+//!     fn close(&mut self) {
+//!         self.0.push_str("z ");
+//!     }
+//! }
+//!
+//! let mut path = SvgPath::default();
+//!
+//! // Scale an outline for glyph 20 and invoke the appropriate methods
+//! // to build an SVG path.
+//! scaler.outline(GlyphId::new(20), &mut path);
+//!
+//! // Print our pretty new path.
+//! println!("{}", path.0);
+//! # }
+//! ```
+//!
+//! The pen based interface is designed to be flexible. Output can be sent directly to a
+//! software rasterizer for scan conversion, converted to an owned path representation (such as
```

Comment:
just a casual note, but I find the line breaking in this file makes it hard to read the actual diff, since many lines are longer and get soft wrapped. I have my editor set to wrap at 80-90 columns and I find that's a reasonable compromise, maybe something like that would be helpful?

Reply:
Apologies! Aside from lengthy URLs, I do try to keep comment widths to roughly 80 characters but it seems to have gotten away from me here. I'll reflow these.